src/components/pages/dashboard.tsx

```diff
@@ -13,10 +15,12 @@ export const Dashboard = () => {
 	const [leftSideWidth] = useResize({ direction: "horizontal", initial: 70, max: 70, min: 30, id: resizeId });
 	const { isMobile } = useWindowDimensions();
 	const { getProjectsList, isLoadingProjectsList, projectsList } = useProjectStore();
+	const location = useLocation();
+	const state = location.state as { organizationId?: string };
 
 	useEffect(() => {
-		if (!projectsList.length && isMobile) {
-			getProjectsList();
+		if ((!projectsList.length && isMobile) || state?.organizationId) {
+			getProjectsList(state?.organizationId);
```

Comment:
If you load the project in the `switchOrganization.tsx` page, you don't need this.
```suggestion
		if (!projectsList.length && isMobile) {
			getProjectsList();
```